test/runtests.jl

```diff
@@ -1,11 +1,18 @@
 using Test, Documenter, DynamicObjects
-doctest(DynamicObjects)
 
-Rectangle = DynamicObject{:rectangle}
-rectangle(height, width) = Rectangle((height=height, width=width))
-area(what::Rectangle) = what.height * what.width
-rect = rectangle(10, 20)
+@testset "doctest" begin
+    doctest(DynamicObjects)
+end
+
+module Foo
+    using DynamicObjects
+    Rectangle = DynamicObject{:rectangle}
+    rectangle(height, width) = Rectangle((height=height, width=width))
+    area(what::Rectangle) = what.height * what.width
+    rect = rectangle(10, 20)
+end
 
 @testset "DynamicObjects.jl" begin
-    @test rect.area == 200
+    @test Foo.area(Foo.rect) == 200
+    @test_broken Foo.rect.area == 200
```

Comment:
@nsiccha this is the bit that's currently not working (due to the explicit reference to `Main` in https://github.com/nsiccha/DynamicObjects.jl/blob/main/src/DynamicObjects.jl#L76)

Reply:
Yeah, just ran into this as well, currently trying to figure out what's the right way to resolve this :)